sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_client.py

```diff
@@ -0,0 +1,480 @@
+    using a custom transport will override and remove the following functionality:
+        1) Authentication using a client certificate.
+        2) TLS verification using the Confidential Ledger TLS certificate.
+
+    :param str ledger_url: URL of the Confidential Ledger service.
+    :param credential: A credential object for authenticating with the Confidential Ledger.
+    :type credential: ~azure.confidentialledger.ConfidentialLedgerCertificateCredential
+    :param str ledger_certificate_path: The path to the ledger's TLS certificate.
+    :keyword api_version: Version of the Confidential Ledger API to use. Defaults to the most recent.
+    :type api_version: ~azure.confidentialledger.ApiVersion
+    """
+
+    def __init__(self, ledger_url, credential, ledger_certificate_path, **kwargs):
+        # type: (str, Union[ConfidentialLedgerCertificateCredential, TokenCredential], str, Any) -> None
+
+        super(ConfidentialLedgerClient, self).__init__(
+            ledger_url=ledger_url,
+            credential=credential,
+            ledger_certificate_path=ledger_certificate_path,
+            **kwargs,
+        )
+
+    @distributed_trace
+    def append_to_ledger(
+        self,
+        entry_contents,  # type: str
+        **kwargs,  # type: Any
+    ):
+        # type: (...) -> AppendResult
+        """Appends an entry to the Confidential Ledger.
+
+        :param entry_contents: Text to write to the ledger.
+        :type entry_contents: str
+        :keyword str sub_ledger_id: Identifies the sub-ledger to append to. If none is
+            specified, the service will use the service-default sub-ledger id.
+        :keyword bool wait_for_commit: If True, this method will not return until the write is
+            durably saved to the ledger.
+        """
+
+        sub_ledger_id = kwargs.pop("sub_ledger_id", None)
+        wait_for_commit = kwargs.pop("wait_for_commit", False)
+
+        if entry_contents is None:
+            raise ValueError("entry_contents must not be a string")
+
+        result = self._client.post_ledger_entry(
+            ledger_base_url=self._ledger_url,
+            contents=entry_contents,
+            sub_ledger_id=sub_ledger_id,
+            cls=kwargs.pop("cls", AppendResult._from_pipeline_result),
+            **kwargs,
+        )
+
+        if wait_for_commit:
+            self.wait_until_durable(result.transaction_id, **kwargs)
+
+        return result
+
+    @distributed_trace
+    def create_or_update_user(
+        self,
+        user_id,  # type: str
+        role,  # type: LedgerUserRole
+        **kwargs,  # type: Any
+    ):
+        # type: (...) -> LedgerUser
+        """Creates a new Confidential Ledger user, or updates an existing one.
+
+        :param user_id: Identifies the user to delete. This should be an AAD object id or
+            certificate fingerprint.
+        :type user_id: str
+        :param role: Role to assigned to the user.
+        :type role: LedgerUserRole
```

Comment:
Should be `str or LedgerUserRole`. Additionally, the type hint on line 107 should be `Union[str, LedgerUserRole]`. Our str enums are basically strings, so users can pass in a string as well